integration-tests/spring-data-jpa/src/main/java/io/quarkus/it/spring/data/jpa/BookRepository.java

```diff
@@ -22,4 +23,14 @@
     boolean existsBookByPublicationYearBetween(Integer start, Integer end);
 
     Optional<Book> findByPublicationYear(Integer year);
+
+    // issue 6205
+    @Query(value = "SELECT publicationYear as publicationYear, COUNT(*) as count FROM Book GROUP BY publicationYear")
```

Comment:
I don't know how it's done in Spring Data, but why can't field name without alias be treated as is? Now `publicationYear as publicationYear` seems redundant.

Reply:
I can try using the original name if it is otherwise missing. Will see where that gets me

Reply:
Yeah I agree with @gytis . From an implementation perspective it shouldn't be any harder (except probably for a more complex regex)

Reply:
agree